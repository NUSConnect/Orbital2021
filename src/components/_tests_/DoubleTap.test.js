import * as React from 'react'
import DoubleTap from '../DoubleTap'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

jest.useFakeTimers()
jest.mock('react-native-vector-icons')

it('renders correctly', () => {
  const tree = renderer.create(<DoubleTap />)
  expect(tree).toMatchSnapshot()
})

it('on tap does not activate', () => {
  const mockTapFn = jest.fn()
  const { getByTestId } = render(<DoubleTap onPress={mockTapFn} />)
  fireEvent.press(getByTestId('doubletap'))
  expect(mockTapFn).not.toHaveBeenCalled()
})
