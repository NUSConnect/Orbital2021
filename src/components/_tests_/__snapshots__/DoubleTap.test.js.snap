// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="doubletap"
/>
`;
